overlord/devicestate/firstboot.go

```diff
@@ -443,6 +437,14 @@ func (m *DeviceManager) populateStateFromSeedImpl(tm timings.Measurer) ([]*state
 	markSeeded.WaitAll(ts)
 	markSeeded.WaitAll(endTs)
 	endTs.AddTask(markSeeded)
+
+	// it is important that mark-seeded be in these lanes. otherwise, the tasks
+	// that are created by hooks to restart services will not properly be
+	// inserted in the chain of tasks before mark-seeded. see
+	// prepareQueueCommand in hookstate/ctlcmd.
```

Comment:
should this be clarified a bit as endTs contains also the validation-set tracking task? I also wonder if that one should or shouldn't be add to snapstate.FinalTasks, do we want services to run after or before it?